bundles/org.openhab.core.io.websocket/src/main/java/org/openhab/core/io/websocket/audiopcm/PCMWebSocketAudioSource.java

```diff
@@ -0,0 +1,200 @@
+ */
+@NonNullByDefault
+public class PCMWebSocketAudioSource implements AudioSource {
+    private final Logger logger = LoggerFactory.getLogger(PCMWebSocketAudioSource.class);
+    public static int supportedBitDepth = 16;
+    public static int supportedSampleRate = 16000;
+    public static int supportedChannels = 1;
+    public static AudioFormat supportedFormat = new AudioFormat(AudioFormat.CONTAINER_WAVE,
+            AudioFormat.CODEC_PCM_SIGNED, false, supportedBitDepth, null, (long) supportedSampleRate,
+            supportedChannels);
+    private final String sourceId;
+    private final String sourceLabel;
+    private final PCMWebSocketConnection websocket;
+    private @Nullable PipedOutputStream sourceAudioPipedOutput;
+    private @Nullable PipedInputStream sourceAudioPipedInput;
+    private @Nullable InputStream sourceAudioStream;
+    private final PipedAudioStream.Group streamGroup = PipedAudioStream.newGroup(supportedFormat);
+    private @Nullable Future<?> sourceWriteTask;
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool("pcm-audio-source");
+    private byte @Nullable [] streamId;
+
+    public PCMWebSocketAudioSource(String id, String label, PCMWebSocketConnection websocket) {
+        this.sourceId = id;
+        this.sourceLabel = label;
+        this.websocket = websocket;
+    }
+
+    @Override
+    public String getId() {
+        return this.sourceId;
+    }
+
+    @Override
+    public String getLabel(@Nullable Locale locale) {
+        return this.sourceLabel;
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return Set.of(supportedFormat);
+    }
+
+    @Override
+    public AudioStream getInputStream(AudioFormat audioFormat) throws AudioException {
+        try {
+            final PipedAudioStream stream = streamGroup.getAudioStreamInGroup();
+            synchronized (streamGroup) {
+                if (this.streamGroup.size() == 1) {
+                    logger.debug("Send start listening {}", getId());
+                    this.streamId = null;
+                    websocket.setListening(true);
+                }
+            }
+            stream.onClose(this::onStreamClose);
+            return stream;
+        } catch (IOException e) {
+            throw new AudioException(e);
+        }
+    }
+
+    public void close() throws Exception {
+        streamGroup.close();
+    }
+
+    public void writeToStreams(byte[] id, int sampleRate, int bitDepth, int channels, byte[] payload) {
+        if (streamGroup.isEmpty()) {
+            logger.debug("Source already disposed ignoring data");
+            return;
+        }
+        if (this.streamId == null) {
+            this.streamId = id;
+        } else if (!Arrays.equals(this.streamId, id)) {
+            // source is only ready to handle a single data line
+            logger.warn("Ignoring data from source stream {}", id);
```

Comment:
Probably add the information that source can only handle a single data line to the log statement.